cpp/src/arrow/engine/substrait/util.h

```diff
@@ -23,6 +23,7 @@
 #include "arrow/compute/registry.h"
 #include "arrow/engine/substrait/api.h"
 #include "arrow/engine/substrait/options.h"
+#include "arrow/engine/substrait/util_internal.h"
```

Comment:
You can't import `util_internal.h` from any public header file.  Instead, import this directly from `relation_internal.cc`, `expression_internal.cc`, (any .cc file is ok, even if not marked internal), etc.

Reply:
Got it.